gpbft/participant.go

```diff
@@ -145,6 +145,8 @@ func (p *Participant) ReceiveAlarm() (err error) {
 
 func (p *Participant) beginInstance() error {
 	chain, power, beacon := p.host.GetCanonicalChain()
+	// Limit length of the chain to be proposed.
+	chain = chain.Prefix(CHAIN_MAX_LEN - 1)
```

Comment:
The chain includes the base, right? This should be `CHAIN_MAX_LEN` in that case (no `-1`).

Reply:
(it would be _nice_ to assert that the base matches the end of our previous proposal...

Reply:
The `Prefix` method adds one: `Prefix(0)` is the base chain. Yes perhaps this isn't the ideal semantics, but it makes more sense at some other call sites. I wouldn't object to you overhauling that in another PR, but it's below my threshold at the moment.

Reply:
Or did you mean that the max length should be CHAIN_MAX_LEN tips plus one for the base?

Reply:
Ah, no. I misunderstood prefix. LGTM.